### PR TITLE
Feature/groups UI

### PR DESCRIPTION
**Pull request by Voxanimus**

### CODE REVIEW

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/UserGroupAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class UserGroupAdapter(context: Context,
+    private val groupViewModel: GroupViewModel
+): BaseAdapter() {
+    private var dataSource: List<UserGroup> = listOf()
+    private val inflater: LayoutInflater =
+        context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+
+
+    fun setGroups(groups: Map<GroupID, UserGroup>){
+        dataSource = groups.values.toList()
```
Comment (felko) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/UserGroupAdapter.kt:28]: It seems that you need to notify the adapter that the data has changed using `notifyDataSetChanged()` just after updating `dataSource`.

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/UserGroupAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import com.github.multimatum_team.multimatum.R
+import com.github.multimatum_team.multimatum.model.GroupID
+import com.github.multimatum_team.multimatum.model.UserGroup
+import com.github.multimatum_team.multimatum.viewmodel.GroupViewModel
+import dagger.hilt.EntryPoint
```
Comment (felko) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/UserGroupAdapter.kt:15]: Imports not used

File: app/src/main/res/layout/activity_groups.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (felko) [inline app/src/main/res/layout/activity_groups.xml:1]: Would you mind adding a bit of padding around the group list?

**Review by felko — CHANGES_REQUESTED** [review]

**Review by felko — APPROVED** [review]
Looks good, thanks for helping me on the groups feature.
Please wait for my PR to be merged before merging yours.

File: app/src/test/java/com/github/multimatum_team/multimatum/GroupActivityTest.kt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.github.multimatum_team.multimatum
+
+class GroupActivityTest {
+}
```
Comment (LennyDZ) [inline app/src/test/java/com/github/multimatum_team/multimatum/GroupActivityTest.kt:4]: Maybe add a todo tag, to be sure you'll do these test

**Review by LennyDZ — APPROVED** [review]
There is for sure way too many file. I tried to spot big mistakes, but it's too large to be exhaustive.

**Pull request merged**